spec/factories/authentication.rb

```diff
@@ -35,4 +35,87 @@
     auth_key    nil
     status      "SomeMockedStatus"
   end
+
+  factory :authentication_allow_all do
```

Comment:
There is quite a bit of duplication in these factories. 

The mechanism to avoid this is called [FactoryGirl traits](https://github.com/thoughtbot/factory_girl/blob/master/GETTING_STARTED.md#traits)

```
trait :github do
  type "AuthenticationGithub"
end

FactoryGirl.create(:authentication, :github).userid # => "testuser"
```

Here it looks like we don't use traits effectively and instead make child factories like `authentication_ssh_keypair_root`. I'd just continue following that pattern here and then someone can do a complete refactor later if traits are desired (at least removes duplication):

```
factory :authentication_github, :parent => :authentication do
  type "AuthenticationGithub"
end
```